libsolidity/formal/SMTEncoder.cpp

```diff
@@ -155,8 +155,10 @@ void SMTEncoder::visitFunctionOrModifier()
 
 	if (m_modifierDepthStack.back() == static_cast<int>(function.modifiers().size()))
 	{
+		pushPathCondition(currentPathConditions());
```

Comment:
What if you put the `push/pop` around the outer if/else here, would that work?

Reply:
Probably yes. But I think I prefer to keep it around the call to the analysis for the function statement. I could even push it deeper inside the `if (function.isImplemented())`. Or why would you like to have the guard outside?